lib/puppet/provider/scheduled_task/win32_taskscheduler.rb

```diff
@@ -401,6 +405,10 @@ def validate_trigger(value)
 
   private
 
+  def canonical_command(command)
+    File.expand_path(command).gsub(/\//, '\\')
```

Comment:
I've always thought `expand_path` was a good way to canonicalize paths in ruby, but it recently came to light that it expands `~` based on the user's `HOME`, which isn't good. Can Windows paths contains `~`? If so, we might want to use `Pathname` to canonicalize?